plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridProcedure.java

```diff
@@ -0,0 +1,140 @@
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.dbeaver.model.meta.PropertyLength;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.rdb.DBSProcedureType;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CubridProcedure extends GenericProcedure
+{
+    private List<CubridProcedureParameter> pro_columns;
+    private String returnType;
+
+    public CubridProcedure(
+            @NotNull GenericStructContainer container,
+            @NotNull String procedureName,
+            @Nullable String description,
+            @NotNull DBSProcedureType procedureType,
+            @NotNull String target,
+            @NotNull String returnType) {
+        super(container, procedureName, description, procedureType, null, true);
+        this.returnType = returnType;
+    }
+
+    @NotNull
+    @Override
+    @Property(viewable = true, order = 1)
+    public String getName() {
+        return super.getName();
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 2)
+    public GenericSchema getSchema() {
+        return super.getSchema();
+    }
+
+    @Nullable
+    @Override
+    public GenericCatalog getCatalog() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public GenericFunctionResultType getFunctionResultType() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public GenericPackage getPackage() {
+        return null;
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 20)
+    public String getReturnType() {
+        return returnType;
+    }
+
+    @Nullable
+    @Property(viewable = true, length = PropertyLength.MULTILINE, order = 100)
+    public String getDescription() {
+        return super.getDescription();
+    }
+
+    public void addColumn(@NotNull CubridProcedureParameter column) {
+        if (this.pro_columns == null) {
+            this.pro_columns = new ArrayList<>();
+        }
+        this.pro_columns.add(column);
+    }
+
+    @Nullable
+    public List<CubridProcedureParameter> getParams(@NotNull DBRProgressMonitor monitor) throws DBException {
+        if (pro_columns == null) {
+            loadProcedureColumns(monitor);
+            if (pro_columns == null) {
+                pro_columns = new ArrayList<>();
+            }
+        }
+        return pro_columns;
+    }
+
+    @NotNull
+    @Override
+    public String getFullyQualifiedName(@NotNull DBPEvaluationContext context) {
+        return getName();
+    }
+
+    @Override
+    public void loadProcedureColumns(@NotNull DBRProgressMonitor monitor) throws DBException {
+        try (JDBCSession session = DBUtils.openMetaSession(monitor, getDataSource(), "Read procedure parameter")) {
+            String stmt = "select * from db_stored_procedure_args where sp_name = '" + getName() + "'";

```

Comment:
1. What if you have a procedure with an equal name in another schema? 
2. Please use parameters instead of adding a direct name:
`dbStat.setString(1, getName());`

Reply:
Our procedure doesn't support multi-schema.

Reply:
Ok, thanks